pkg/util/push/otel.go

```diff
@@ -150,10 +153,43 @@ func OTLPHandler(
 		)
 
 		req.Timeseries = metrics
+
+		if EnableOtelMetadataStorage {
+			metadata := otelMetricsToMetadata(otlpReq.Metrics())
+			req.Metadata = metadata
+		}
+
 		return body, nil
 	})
 }
 
+func otelMetricsToMetadata(md pmetric.Metrics) []*mimirpb.MetricMetadata {
+	var metadata []*mimirpb.MetricMetadata
```

Comment:
One small performance improvement here would be to pre-allocate a slice large enough to hold all of the expected metadata for this request.

Reply:
updated